ruby_event_store-active_record/Rakefile

```diff
@@ -0,0 +1,14 @@
+require_relative "lib/ruby_event_store/active_record/generators/migration_generator"
+
+namespace :g do
+  desc "Generate migration"
+  task :migration, :data_type, :migration_path do |_,args|
+    data_type = args[:data_type] || raise("Specify data type (binary, json, jsonb): rake g:migration[data_type]")
+
+    path = RubyEventStore::ActiveRecord::MigrationGenerator
+      .new
+      .call(data_type, args[:migration_path] || "db/migrate")
+
+    puts "Migration #{path} created"
+  end
+end
```

Comment:
Let me think about public API a bit more. We definitely not want to have `:g` namespace.

Reply:
I think Joel captured nice API here: https://railseventstore.org/docs/v2/without_rails/#rom-sequel-migrations

Which I've probably failed to communicate properly earlier.

In short:

* we provide Rake task, but do not depend on `rake` gem
* if one wants to use our migration they are expected to have `rake` and define `Rakefile` with our task imported (in Rails, rake tasks from gems are loaded provided they're in `lib/tasks` but we don't need it for Rails)
* passing options via ENV=var is more pleasant that figuring out Rake API for passing arguments — no one remembers it

Reply:
https://github.com/RailsEventStore/rails_event_store/blob/master/contrib/ruby_event_store-rom/lib/ruby_event_store/rom/rake_task.rb
https://github.com/RailsEventStore/rails_event_store/blob/master/contrib/ruby_event_store-rom/lib/ruby_event_store/rom/tasks/migration_tasks.rake

Perhaps we'll need to polyfill some Rake tasks that ROM exposes but ActiveRecord not, since they mighr be Rails ones. Or maybe everything needed is already shipped with AR gem:
https://github.com/rails/rails/blob/ade27e5028ccbaaa5db8d966500844017925a751/activerecord/lib/active_record/tasks/database_tasks.rb#L38

Reply:
My proposition:
https://github.com/RailsEventStore/rails_event_store/pull/1479/files#diff-321cc0a9c933cf4b29d7879c466f938c7a4c7c81a4d9e1b2fa09d6664f4a5646

https://github.com/RailsEventStore/rails_event_store/pull/1479/files#diff-97e5809164d35afd58d70d31e5c4a606534e82dc2dc600df98a4156fbe7624e1

In Off-Rails project with Rakefile
`require "ruby_event_store/active_record/rake_task"`
the
`bundle exec rake -T`
produces
```
rake db:create              # Creates the database from DATABASE_URL or config/database.yml for the current RAILS_ENV (use db:create:all to create all databases in the config). Without RAILS_ENV or when RAILS_ENV is development, it defaults...
rake db:drop                # Drops the database from DATABASE_URL or config/database.yml for the current RAILS_ENV (use db:drop:all to drop all databases in the config). Without RAILS_ENV or when RAILS_ENV is development, it defaults to dr...
rake db:encryption:init     # Generate a set of keys for configuring Active Record encryption in a given environment
rake db:environment:set     # Set the environment value for the database
rake db:fixtures:load       # Loads fixtures into the current environment's database
rake db:migrate             # Migrate the database (options: VERSION=x, VERBOSE=false, SCOPE=blog)
rake db:migrate:down        # Runs the "down" for a given migration VERSION
rake db:migrate:redo        # Rolls back the database one migration and re-migrates up (options: STEP=x, VERSION=x)
rake db:migrate:status      # Display status of migrations
rake db:migrate:up          # Runs the "up" for a given migration VERSION
rake db:migrations:copy     # Generate migration
rake db:prepare             # Runs setup if database does not exist, or runs migrations if it does
rake db:reset               # Drops and recreates all databases from their schema for the current environment and loads the seeds
rake db:rollback            # Rolls the schema back to the previous version (specify steps w/ STEP=n)
rake db:schema:cache:clear  # Clears a db/schema_cache.yml file
rake db:schema:cache:dump   # Creates a db/schema_cache.yml file
rake db:schema:dump         # Creates a database schema file (either db/schema.rb or db/structure.sql, depending on `ENV['SCHEMA_FORMAT']` or `config.active_record.schema_format`)
rake db:schema:load         # Loads a database schema file (either db/schema.rb or db/structure.sql, depending on `ENV['SCHEMA_FORMAT']` or `config.active_record.schema_format`) into the database
rake db:seed                # Loads the seed data from db/seeds.rb
rake db:seed:replant        # Truncates tables of each database for current environment and loads the seeds
rake db:setup               # Creates all databases, loads all schemas, and initializes with the seed data (use db:reset to also drop all databases first)
rake db:version             # Retrieves the current schema version number
```